src/openeo_test_suite/tests/processes/metadata/test_process_metadata.py

```diff
@@ -0,0 +1,163 @@
+import json
+import pytest
+import requests
+from openeo_test_suite.lib.process_registry import ProcessData
+from openeo_test_suite.lib.process_selection import get_selected_processes
+from openeo_test_suite.lib.backend_under_test import (
+    get_backend_under_test,
+)
+import warnings
+
+
+@pytest.fixture(scope="module")
+def api_processes():
+    endpoint_path = "processes"
+    base_url = get_backend_under_test().connection.root_url
+    if not base_url:
+        raise ValueError("No backend URL configured")
+    if base_url.endswith("/"):
+        base_url = base_url[:-1]
+    full_endpoint_url = f"{base_url}/{endpoint_path}"
+    response = requests.get(full_endpoint_url)
+    if response.status_code != 200:
+        raise ValueError(
+            f"Failed to get processes from {full_endpoint_url}: {response.content}"
+        )
+    return json.loads(response.content)["processes"]
+
+
+def _get_test_id(val):
+    if isinstance(val, ProcessData):
+        return val.process_id
+
+
+@pytest.mark.parametrize(
+    "expected_process",
+    [p for p in get_selected_processes() if p.metadata],
```

Comment:
If I understand the `ProcessRegistry` refactoring of this PR correctly, I think you don't need the `p.metadata` here anymore (it will always be set by design)